pvlib/shading.py

```diff
@@ -553,3 +553,109 @@ def shaded_fraction1d(
     )
 
     return np.clip(t_asterisk, 0, 1)
+
+
+def martinez_shade_factor(shaded_fraction, N_shaded_blocks, N_total_blocks):
```

Comment:
Typically pvlib puts the model author at the end of the function name. `shade_factor_martinez`?  Or better yet, `direct_martinez`?

Reply:
Can you give me some insight on why the `direct_` prefix?

Reply:
Inspired by `pvlib.shading.sky_diffuse_passias`, I was thinking that shade models could be named like `f"{irradiance_component}_{author}"`, where `irradiance_component` is `direct`, `sky_diffuse`, etc.

Reply:
Ah! That makes sense. My bad, I didn't update the example accordingly to reflect the modifier is on an irradiance component and not on the output power - a confusion I had some time ago.